src/setindex.jl

```diff
@@ -31,3 +32,6 @@ end
 @inline setindex(nt::NamedTuple, v, idx::Tuple{Vararg{Symbol}}) = merge(nt, NamedTuple{idx}(v))
 
 @inline setindex(p::Pair, v, idx::Integer) = Pair(setindex(Tuple(p), v, idx)...)
+
+@inline setindex(x::Number, v, idx::Integer) = (@boundscheck @assert idx == 1; v)
```

Comment:
I think it is better to throw a `BoundsError`.
```suggestion
@inline function setindex(x::Number, v, idx::Integer)
    @boundscheck if idx != only(eachindex(x))
        throw(BoundsError(x, idx))
    end
    return v
end
```